python/modules/IcePy/PropertiesAdmin.cpp

```diff
@@ -119,8 +57,36 @@ nativePropertiesAdminAddUpdateCB(NativePropertiesAdminObject* self, PyObject* ar
         return 0;
     }
 
-    (*self->callbacks).push_back(new UpdateCallbackWrapper(callback));
-    (*self->admin)->addUpdateCallback((*self->callbacks).back());
+    std::function<void()> remover = (*self->admin)->addUpdateCallback(
+        [callback](const Ice::PropertyDict& dict)
+        {
+            AdoptThread adoptThread; // Ensure the current thread is able to call into Python.
+
+            PyObjectHandle result = PyDict_New();
+            if (result.get())
+            {
+                for (Ice::PropertyDict::const_iterator p = dict.begin(); p != dict.end(); ++p)
+                {
+                    PyObjectHandle key = createString(p->first);
+                    PyObjectHandle val = createString(p->second);
+                    if(!val.get() || PyDict_SetItem(result.get(), key.get(), val.get()) < 0)
```

Comment:
It is not clear to me how error reporting works here. I assume `!val.get()` is an error condition, but there is no similar check for `key`

Reply:
I didn't change this code.